File: practicum01/Practicum1A_AntennaPatternSOLUTION.py
import numpy as np
import enthought.mayavi.mlab as mlab
#Script that plots antenna pattern response of pulsar-earth system and LIGO

phi = np.linspace(0,2*np.pi,100)
theta = np.linspace(0,np.pi,100)

# Set up our grid

[phi,theta] = np.meshgrid(phi,theta)

# Radius of displayed sphere

r = 5

x = r*np.sin(theta)*np.cos(phi)
y = r*np.sin(theta)*np.sin(phi)
z = r*np.cos(theta)

#Plot Sphere

mlab.figure("Figure 1")
figure1 = mlab.mesh(x,y,z,representation='wireframe')
mlab.axes()
mlab.title("Figure 1")
mlab.outline()

#Make the radius a function of theta and phi
#antenna pattern response to a +/- polarized wave
#See Eq. 7 of Anholm et al. (Phys. Rev. D 79, 084030 (2009))

rp = np.absolute(.5*np.sin(theta)**2 * (np.cos(phi)**2 - 
	np.sin(phi)**2) / (1 + np.cos(theta)))

rp_mask = np.ma.masked_invalid(rp)

x = rp*np.sin(theta)*np.cos(phi)
x[np.isinf(x)] = np.nan
y = rp*np.sin(theta)*np.sin(phi)
y[np.isinf(y)] = np.nan
z = rp*np.cos(theta)
z[np.isinf(z)] = np.nan


mlab.figure("Figure 2")
figure2 = mlab.mesh(x,y,z,mask=np.ma.getmask(rp_mask),representation='wireframe')
mlab.axes()
mlab.title("Figure 2")
mlab.outline()

# Antenna pattern including frequency dependence (i.e. the pulsar term)
# (Eqs. 16 and 17 or Anholm et al.)

fL=10 #This is typical for pulsars
#Students should try this and notice a frequency dependence in  pictu

rp = np.absolute((np.exp(-2j*np.pi*fL*(1+np.cos(theta))) - 1)*
	.5*np.sin(theta)**2*(np.cos(phi)**2 - np.sin(phi)**2) 
	/ (1 + np.cos(theta)))

rp_mask = np.ma.masked_invalid(rp)

x = rp_mask*np.sin(theta)*np.cos(phi)
y = rp_mask*np.sin(theta)*np.sin(phi)
z = rp_mask*np.cos(theta)

mlab.figure("Figure 3")
figure3 = mlab.mesh(x,y,z,representation='wireframe')
mlab.axes()
mlab.title("Figure 3")
mlab.outline()
# EXCERCISE: At this point they do a Taylor expansion for theta=pi+delta, where delta
# is sufficiebtly small that the exponential can be expanded. This way they
# see where the frequency dependence in the response at the south pole is coming from 

# antenna pattern for LIGO (see Eq. 3 of Phys. Rev. D 79, 082002 (2009))
# this is kind of a cool paper, because it deals with non-Einstein tehories
# of gravity

# plus polarization with the polarization angle set to 0

psi = 0

rp = np.absolute(.5 * (1 + np.cos(theta)**2) * np.cos(2*phi) * np.cos(2*psi)
	- np.cos(theta)*np.sin(2*theta)*np.sin(2*psi))

rp_mask = np.ma.masked_invalid(rp)
x = rp_mask*np.sin(theta)*np.cos(phi)
y = rp_mask*np.sin(theta)*np.sin(phi)
z = rp_mask*np.cos(theta)

mlab.figure("Figure 4")
figure4 = mlab.mesh(x,y,z,representation='wireframe')
mlab.axes()
mlab.title("Figure 4")
mlab.outline()

mlab.show()

File: practicum01/Practicum1A_AntennaPatternSOLUTION.m
%script that plots antenna pattern response of pulsar-earth system and LIGO

% the idea here is we give the students a script that plots a meshed sphere and ask them to
% edit it to show the pulsar earth system response
clear

phi=linspace(0,2*pi,100);
theta=linspace(0,pi,100);

% set up grid
[phi,theta]=meshgrid(phi,theta);

% radius of sphere
r=5;

x=r.*sin(theta).*cos(phi);
y=r.*sin(theta).*sin(phi);
z=r.*cos(theta);

% plot sphere
figure; mesh(x,y,z)

% they just need to make the radius a function of theta and phi
% antenna pattern response to a +-polarized wave
% Eq. 7 of Anholm et al. (Phys. Rev. D 79, 084030 (2009))
rp=abs(1/2*sin(theta).^2 .*(cos(phi).^2-sin(phi).^2) ./ (1+cos(theta)));

x=rp.*sin(theta).*cos(phi);
y=rp.*sin(theta).*sin(phi);
z=rp.*cos(theta);

figure; mesh(x,y,z)

% antenna pattern including frequency dependence (i.e. the pulsar term)
% (Eqs. 16 and 17 of Anholm et al.)

fL=10; % this is typical for pulsars
%fL=1; % for students to try. They should notice a frequency dependence in
%the picture

rp=abs(    (exp(-2*pi*i*fL*(1+cos(theta)))-1) .*...
    1/2*sin(theta).^2 .*(cos(phi).^2-sin(phi).^2)./ (1+cos(theta)));

x=rp.*sin(theta).*cos(phi);
y=rp.*sin(theta).*sin(phi);
z=rp.*cos(theta);

figure; mesh(x,y,z)

% EXCERCISE: At this point they do a Taylor expansion for theta=pi+delta, where delta
% is sufficiebtly small that the exponential can be expanded. This way they
% see where the frequency dependence in the response at the south pole is coming from 

% antenna pattern for LIGO (see Eq. 3 of Phys. Rev. D 79, 082002 (2009))
% this is kind of a cool paper, because it deals with non-Einstein tehories
% of gravity

% plus polarization with the polarization angle set to 0

psi=0;

rp=abs( 1/2 * (1+cos(theta).^2) .* cos(2*phi) .* cos(2*psi)...
    -cos(theta).*sin(2*theta)*sin(2*psi));

x=rp.*sin(theta).*cos(phi);
y=rp.*sin(theta).*sin(phi);
z=rp.*cos(theta);

figure; mesh(x,y,z)






